src/core/internal/convert.d

```diff
@@ -238,14 +281,14 @@ private template FloatTraits(T) if(floatFormat!T == FloatFormat.DoubleDouble) //
     enum NINF     = Float(0, 0x7ff, 1);
 }
 
-private template FloatTraits(T) if(floatFormat!T == FloatFormat.Quadruple) //Unsupported in CTFE
+private template FloatTraits(T) if(floatFormat!T == FloatFormat.Quadruple)
 {
     enum EXPONENT = 15;
     enum MANTISSA = 112;
     enum ZERO     = Float(0, 0, 0);
     enum NZERO    = Float(0, 0, 1);
-    enum NAN      = Float(-1, 0x7fff, 0);
-    enum NNAN     = Float(-1, 0x7fff, 1);
+    enum NAN      = Float(0, 0x7fff, 0, 0x80000000000000UL);
+    enum NNAN     = Float(0, 0x7fff, 1, 0x80000000000000UL);
```

Comment:
I didn't bother adding `0`'s everywhere else the `Float` constructor was called, as this is the only 112-bit mantissa where the most significant bits in `mantissa2` needed to be set.